test/Common.FunctionalTests/AppOfflineTests.cs

```diff
@@ -173,6 +158,44 @@ public class AppOfflineTests : IISFunctionalTestBase
             await AssertRunning(deploymentResult);
         }
 
+        [ConditionalTheory]
+        [InlineData(HostingModel.InProcess)]
+        [InlineData(HostingModel.OutOfProcess)]
+        public async Task AppOfflineAddedAndRemovedStress(HostingModel hostingModel)
+        {
+            var deploymentResult = await AssertStarts(hostingModel);
+
+            var load = Helpers.StressLoad(deploymentResult.HttpClient, "/HelloWorld", response => {
+                var statusCode = (int)response.StatusCode;
+                Assert.True(statusCode == 200 || statusCode == 503, "Status code was " + statusCode);
+            });
+
+            for (int i = 0; i < 100; i++)
+            {
+                // AddAppOffline might fail if app_offline is being read by ANCM and deleted at the same time
+                RetryHelper.RetryOperation(
+                    () => AddAppOffline(deploymentResult.DeploymentResult.ContentRoot),
+                    e => Logger.LogError($"Failed to create app_offline : {e.Message}"),
+                    retryCount: 3,
+                    retryDelayMilliseconds: 100);
```

Comment:
We may want to add variants that send a request in between adding and removing app offline.

Reply:
Wouldn't it be AppOfflineDroppedWhileSiteRunning_SiteShutsDown_InProcess and AppOfflineDroppedWhileSiteRunning_SiteShutsDown_OutOfProcess?